clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -544,6 +554,47 @@ LogicalResult VecTernaryOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// VecShuffleInts
+//===----------------------------------------------------------------------===//
+
+LogicalResult VecShuffleIntsOp::verify() {
+  // The number of elements in the indices array must match the number of
+  // elements in the result type.
```

Comment:
There's a trait that helps with this: `AttrSizedOperandSegments`. No need to address for this PR, but it would be nice if we can use it.

Reply:
I will look into `AttrSizedOperandSegments`.  Thanks for the suggestion.

Reply:
`AttrSizedOperandSegments` can only be used when the operation has multiple variadic or optional operands.  (See `generateNamedOperandGetters` in `mlir/tools/mlir-tblgen/OpDefinitionsGen.cpp`)  `VecShuffleOp` has only one variadic operand, so giving it that attribute would result in a validation error.